### PR TITLE
Keep original filename for SEO

### PR DESCRIPTION
**Pull request by epidoux**
Hi Can,

Just made some changes to handle the case where you want to keep the original filename (for SEO).
Let me know your thoughts!

Great bundle by the way ;)

Cheers,

Eric

### CODE REVIEW

**Comment by comur** [comment]
Hi @epidoux,

I think it's a good idea but you need to check if file already exists and if yes need to change the name (by adding an int maybe, like filename-1.ext filename-2.ext) in order to not overwrite existing one.

**Comment by epidoux** [comment]
yep, I forgot. I added a timestamp before the filename to handle this. Is it ok for you or do you want an int?

**Comment by comur** [comment]
Nope it's ok for me but you can add a "-" between timestamp and the filename ? And use time() instead of date('U') (it's the same thing but better for perf i guess).
And can you use { and } for if that you added please ? for better readability.

Thanks anyway !

**Comment by epidoux** [comment]
ok done. Let me know if there is something else. 
I will try to work on other issues when I will have time!

File: Controller/UploadController.php [hunk 1]
```diff
@@ -33,10 +35,15 @@ public function uploadImageAction(Request $request
         
         $webDir = $config['uploadConfig']['webDir'];
         $webDir = substr($webDir, -strlen('/')) === '/' ? $webDir : $webDir . '/';
-        $filename = sha1(uniqid(mt_rand(), true));
-        
-        $thumbsDir = $this->container->getParameter('comur_image.thumbs_dir');
-        $thumbSize = $this->container->getParameter('comur_image.media_lib_thumb_size');
+        if($config['uploadConfig']['generateFilename']) $filename = sha1(uniqid(mt_rand(), true));
```
Comment (comur) [inline Controller/UploadController.php]: Hi @epidoux 

There is just this one left, thanks! And thanks for your help anyway !

**Comment by hemantchaudhari** [comment]
May I know where 'file_name' is being used in UploadHandler class? 'file_name' is passed in $handlerConfig in UploadController.php, but not used inside UploadHandler class.

**Comment by georgesamy** [comment]
Nice feature, I actually need it myself. Is it getting merged? @comur

**Comment by comur** [comment]
Hi @georgesamy 
Can you test this PR on your project before I can merge it (or someone else ?) because I don't have any project to integrate and test it for now.

**Comment by georgesamy** [comment]
I have tested it and it is working fine. In fact, I have forked the repo and merged this feature for my build.

**Comment by comur** [comment]
Ok thanks so I can merge it on official branch. You can put back dev-master in your project

**Comment by georgesamy** [comment]
Thank you!

**Comment by epidoux** [comment]
Great, thanks!

**Pull request merged**